engine/function/src/templates/Numeric.ftl

```diff
@@ -1055,14 +1079,18 @@ public class Numeric {
         ${pt.primitive} val = ${pt.maxValue};
         long index = NULL_LONG;
         long count = 0;
-        final long n = values.size();
-
-        for (int i = 0; i < n; i++) {
-            ${pt.primitive} c = values.get(i);
-            if (!isNull(c) && (c < val || (c == val && count == 0) )) {
-                val = c;
-                index = i;
-                count++;
+        long i = 0;
+
+        try (final ${pt.vectorIterator} vi = values.iterator()) {
+            while ( vi.hasNext() ) {
+                final ${pt.primitive} c = vi.${pt.iteratorNext}();
+                if (!isNull(c) && (c < val || (c == val && count == 0) )) {
+                    val = c;
+                    index = i;
+                    count++;
+                }
+
+                i++;
```

Comment:
Sticking this comment on an arbitrary line. We have two `toArray()` issues in this file:
1. L1140, in `median`: We intend to copy, but we might not be
2. L1180, in `percentile`: Same issue.